storagenode/console/consoleserver/server.go

```diff
@@ -127,163 +104,71 @@ func (server *Server) Close() error {
-	if err != nil {
-		return response, err
+	if req.Method != http.MethodGet {
+		http.Error(wr, "method not allowed", http.StatusMethodNotAllowed)
+		return
 	}
 
-	// TODO: uncomment in future, when caching will be implemented
-	// diskSpaceChartData, err := server.getDiskSpaceChartData(ctx, satelliteID, satellites)
-	// if err != nil {
-	// 	return response, err
-	// }
-
-	uptime := server.service.GetUptime(ctx)
-	nodeID := server.service.GetNodeID(ctx)
-
-	// TODO: uncomment in future, when caching will be implemented
-	// if satelliteID != nil {
-	// 	satelliteStats, err := server.service.GetStatsFromSatellite(ctx, *satelliteID)
-	// 	if err != nil {
-	// 		return response, err
-	// 	}
-	//
-	// 	response.UptimeCheck = satelliteStats.UptimeCheck
-	// 	response.AuditCheck = satelliteStats.AuditCheck
-	// }
-
-	response.DiskSpace = *space
-	response.Bandwidth = *usage
-	response.WalletAddress = walletAddress
-	response.VersionInfo = versionInfo
-	response.IsLastVersion = true
-	response.Uptime = uptime
-	response.NodeID = nodeID.String()
-	response.Satellites = satellites
-	response.BandwidthChartData = bandwidthChartData
-	//response.DiskSpaceChartData = diskSpaceChartData
-
-	return response, nil
-}
+	wr.Header().Set(contentType, applicationJSON)
 
-func (server *Server) getBandwidth(ctx context.Context, satelliteID *storj.NodeID) (_ *console.BandwidthInfo, err error) {
-	if satelliteID != nil {
-		return server.service.GetBandwidthBySatellite(ctx, *satelliteID)
+	var data struct {
+		Data  *console.Satellite `json:"data"`
+		Error error              `json:"error"`
 	}
 
-	return server.service.GetUsedBandwidthTotal(ctx)
-}
-
-func (server *Server) getBandwidthChartData(ctx context.Context, satelliteID *storj.NodeID) (_ []console.BandwidthUsed, err error) {
-	from, to := date.MonthBoundary(time.Now().UTC())
-
-	if satelliteID != nil {
-		return server.service.GetDailyBandwidthUsed(ctx, *satelliteID, from, to)
-	}
+	satelliteID := strings.TrimLeft(req.URL.Path, "/api/satellite/")
 
-	return server.service.GetDailyTotalBandwidthUsed(ctx, from, to)
-}
+	data.Data, data.Error = server.getSatelliteData(ctx, satelliteID)
 
-func (server *Server) getStorage(ctx context.Context, satelliteID *storj.NodeID) (_ *console.DiskSpaceInfo, err error) {
-	if satelliteID != nil {
-		return server.service.GetUsedStorageBySatellite(ctx, *satelliteID)
+	if err := json.NewEncoder(wr).Encode(&data); err != nil {
+		server.log.Error(err.Error())
 	}
-
-	return server.service.GetUsedStorageTotal(ctx)
 }
 
-func (server *Server) checkSatelliteID(satelliteIDs storj.NodeIDList, satelliteID storj.NodeID) error {
-	for _, id := range satelliteIDs {
-		if satelliteID == id {
-			return nil
-		}
+// getSatelliteData returns Satellite data based on provided string representation
+// returns consolidated Satellite data if satelliteID is empty string
+func (server *Server) getSatelliteData(ctx context.Context, satelliteID string) (*console.Satellite, error) {
```

Comment:
Let's make separate funcs depending on the satelliteID rather than overloading the same call with string.

Reply:
I have 2 separate func `service.GetSatellite` and `service.GetAllSatellitesData`, and invoking one them depending on `satelliteID` param. Do you mean that we should create one more endpoint for consolidated satellite data?

Reply:
I just mean remove this func and maybe handle the param one level above.

Reply:
Did like that in the first place

But then I have ugly `if/else` block, smth like:
`	satelliteID := strings.TrimLeft(req.URL.Path, "/api/satellite/")
	if satelliteID == "" {
		data.Data, data.Error = server.service.GetAllSatellitesData(ctx)
	} else {
		var id storj.NodeID
		id, data.Error = storj.NodeIDFromString(satelliteID)
		if data.Error == nil {
			data.Data, data.Error = server.service.GetSatelliteData(ctx, id)
		}
	}
`

Reply:
```
mux.Handle("/api/satellite", http.HandlerFunc(server.satellitesHandler))
mux.Handle("/api/satellite/", http.HandlerFunc(server.satelliteHandler))
```

or 

```
mux.Handle("/api/satellites", http.HandlerFunc(server.satellitesHandler))
mux.Handle("/api/satellite/", http.HandlerFunc(server.satelliteHandler))
```

should also work.

Reply:
Also since you are using mux package you can use variables

Reply:
You could also use gorilla mux, which would allow better parsing of path arguments.